components/drafts/channel_draft/channel_draft.tsx

```diff
@@ -65,11 +93,16 @@ function ChannelDraft({
             return;
         }
 
-        dispatch(createPost(post, value.fileInfos));
-        dispatch(removeDraft(id, channel.id));
-
-        history.push(channelUrl);
-    }, [value, channelUrl, user.id, channel.id]);
+        if (
+            postPriorityEnabled &&
+            value?.metadata?.priority?.priority === PostPriority.URGENT &&
+            value?.metadata?.priority?.persistent_notifications
```

Comment:
nit: this is the second time this check appears, so we could create a helper for that